Sanitise EVP_Q_mac with a NULL key
    
This is a legacy compatibility path (for RT3095, the detail of which
has been lost to time). Ensure that this continues to work in the
presence of a provider that enforces a strict minimum key size (by
choosing a key length that should exceed any reasonable minimum).

Index: openssl/crypto/evp/mac_lib.c
===================================================================
--- openssl.orig/crypto/evp/mac_lib.c
+++ openssl/crypto/evp/mac_lib.c
@@ -287,8 +287,11 @@ unsigned char *EVP_Q_mac(OSSL_LIB_CTX *l
         subalg_param[0] = OSSL_PARAM_construct_utf8_string(param_name, (char *)subalg, 0);
     }
     /* Single-shot - on NULL key input, set dummy key value for EVP_MAC_Init. */
-    if (key == NULL && keylen == 0)
-        key = data;
+    if (key == NULL && keylen == 0) {
+        static const char dummy_key[32] = {0};
+        key = dummy_key;
+        keylen = sizeof(dummy_key);
+    }
     if ((ctx = EVP_MAC_CTX_new(mac)) != NULL
         && EVP_MAC_CTX_set_params(ctx, subalg_param)
         && EVP_MAC_CTX_set_params(ctx, params)
